cmd/internal/cli/build_linux.go

```diff
@@ -510,7 +510,13 @@ func processDefs(args []string, argFile string, defs []types.Definition) ([]type
 	diff := provideKeys.Difference(matchedKeys)
 	if !diff.Empty() {
 		vars := strings.Fields(strings.TrimPrefix(diff.String(), "HashSet"))
-		return defs, fmt.Errorf("while matching all variables, there are unmatched %s variables", strings.Join(vars, ""))
+
+		// All mismatched keys are in provided keys
+		if buildArgs.buildArgsUnusedWarn {
+			sylog.Warningf("While matching all variables, there are unmatched %s variables", strings.Join(vars, ""))
+		} else {
+			return defs, fmt.Errorf("while matching all variables, there are unmatched %s variables. Use option --warn-unused-build-args to show warnings instead of fatal messages", strings.Join(vars, ""))
```

Comment:
Likewise for the first part of the sentence use `"unused build args: %s."` and put a space between the variables.  Change `warnings` to `a warning` and `fatal messages` to `a fatal message`.

Reply:
thanks! resolved here
https://github.com/apptainer/apptainer/pull/1462/commits/93029198be71cad69e4c5391cf6514ccbebae11f